web/src/app/forms/FormContainer.js

```diff
@@ -117,50 +84,55 @@ export class FormContainer extends React.PureComponent {
         return i !== parseInt(idx, 10)
       })
 
-      return this.props.onChange(
+      return props.onChange(
         mapOnChangeValue(set(mapValue(oldValue), arrayPath, newArr)),
       )
     }
 
-    return this.props.onChange(
+    return props.onChange(
       mapOnChangeValue(set(mapValue(oldValue), fieldName, value)),
     )
   }
 
-  render() {
-    return <FormContext.Consumer>{this.renderComponent}</FormContext.Consumer>
-  }
-
-  renderComponent = ({ disabled: formDisabled, addSubmitCheck }) => {
-    const {
-      value,
-      mapValue,
-      optionalLabels,
-      disabled: containerDisabled,
-    } = this.props
-
-    return (
-      <MountWatcher
-        onMount={() => {
-          this._unregister = addSubmitCheck(this.onSubmit)
-        }}
-        onUnmount={() => {
-          this._unregister()
+  return (
+    <FormContext.Consumer>
+      <FormContainerContext.Provider
+        value={{
+          value: mapValue(value),
+          disabled: formDisabled || containerDisabled,
+          errors: validationErrors.concat(props.errors),
```

Comment:
Use the ones from state instead
```suggestion
          errors: persistentValidationErrors(validationErrors).concat(props.errors),
```